front/pages/api/w/[wId]/data_sources/[dsId]/files.ts

```diff
@@ -103,14 +103,49 @@ async function handler(
         { file, upsertArgs: upsertArgs }
       );
       if (rUpsert.isErr()) {
+        let status_code: number;
+        let type: APIErrorType;
+
+        switch (rUpsert.error.code) {
+          case "file_not_ready":
+          case "invalid_file":
+          case "title_too_long":
+          case "invalid_url":
+          case "missing_csv":
+          case "invalid_csv_content":
+          case "invalid_csv_and_file":
+          case "invalid_content_error":
+          case "resource_not_found":
+          case "table_not_found":
+            status_code = 400;
+            type = "invalid_request_error";
+            break;
+
+          case "file_not_found":
```

Comment:
Nope I think that's a 400 here since the file is a body parameter not a route parameter